test/dotnet-publish.Tests/PublishDesktopTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.Tools.Publish.Tests
+{
+    public class PublishDesktopTests : TestBase
+    {
+        [WindowsOnlyTheory]
+        [InlineData("KestrelDesktopWithRuntimes", "http://localhost:20201", null, "libuv.dll", true)]
+        [InlineData("KestrelDesktopWithRuntimes", "http://localhost:20202", "win7-x64", "libuv.dll", true)]
+        [InlineData("KestrelDesktop", "http://localhost:20204", null, "libuv.dll", true)]
+        [InlineData("KestrelDesktop", "http://localhost:20205", "win7-x64", "libuv.dll", true)]
+        public async Task DesktopApp_WithKestrel_WorksWhenPublished(string project, string url, string runtime, string libuvName, bool runnable)
+        {
+            var testInstance = GetTestInstance()
+                .WithLockFiles();
+
+            var publishCommand = new PublishCommand(Path.Combine(testInstance.TestRoot, project), runtime: runtime);
+            var result = await publishCommand.ExecuteAsync();
+
+            result.Should().Pass();
+
+            // Test the output
+            var outputDir = publishCommand.GetOutputDirectory(portable: false);
+            outputDir.Should().HaveFile(libuvName);
+            outputDir.Should().HaveFile(publishCommand.GetOutputExecutable());
+
+            Task exec = null;
+            if (runnable)
+            {
+                var command = new TestCommand(Path.Combine(outputDir.FullName, publishCommand.GetOutputExecutable()));
+                try
+                {
+                    exec = command.ExecuteAsync(url);
+                    NetworkHelper.IsServerUp(url).Should().BeTrue($"Unable to connect to kestrel server - {project} @ {url}");
+                    NetworkHelper.TestGetRequest(url, url);
+                }
+                finally
+                {
+                    command.KillTree();
+                }
+                if (exec != null)
+                {
+                    await exec;
+                }
+            }
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("KestrelDesktop", "http://localhost:20207")]
```

Comment:
If it doesn't have any real functionality what is the purpose of the test? If it's just to make sure the layout is accurate, we can add tests that do that for these Kestrel projects. (Though earlier tests in this file do indeed do a file check before trying to run)